katello/katello-backup

```diff
@@ -54,49 +95,81 @@ def run_cmd(command, exit_codes=[0])
   end
 end
 
-def backup_db_online
-  puts "Backing up postgres online schema... "
-  run_cmd("runuser - postgres -c 'pg_dumpall -g > #{@dir}/pg_globals.dump'")
-  run_cmd("runuser - postgres -c 'pg_dump -Fc foreman > #{@dir}/foreman.dump'")
-  run_cmd("runuser - postgres -c 'pg_dump -Fc candlepin > #{@dir}/candlepin.dump'")
-  puts "Done."
-
-  puts "Backing up mongo online schema... "
-  run_cmd('mongodump --host localhost --out mongo_dump')
-  puts "Done."
+def snapshot_backup
+  snapsize = @options[:snapshot_size] || "2G"
+  mountdir = "#{@options[:snapshot_mount]}/rootsnapshot" || "/mnt/snap/rootsnapshot"
+  FileUtils.mkdir_p mountdir
+  run_cmd("katello-service stop #{EXCLUDED}")
+  run_cmd("lvcreate -s -nrootsnapshot -L#{snapsize} /dev/#{@options[:snapshot]}/root")
+  run_cmd("mount /dev/#{@options[:snapshot]}/rootsnapshot #{mountdir} -txfs -onouuid,ro")
+  run_cmd("katello-service start #{EXCLUDED}")
```

Comment:
Respectfully disagree on this point. Since we aren't stopping the excluded services, we shouldn't attempt to start them, if only for the few seconds of time it may save. If the excluded services should grow, this keeps it more manageable and obvious as well.